File: scripts/analyze_media.py
import os
import subprocess
import json

base_dir = r"C:\Users\javi_\Desktop\Proyectos\UtopiaVanLife\Landing Videos-images\Landing"
preview_dir = r"C:\Users\javi_\Desktop\Proyectos\UtopiaVanLife\UtopiaVanLife_appWb\temp_previews"
os.makedirs(preview_dir, exist_ok=True)

files = []
for root, dirs, filenames in os.walk(base_dir):
    rel_folder = os.path.relpath(root, base_dir)
    for f in filenames:
        files.append((os.path.join(root, f), rel_folder, f))

print(f"Total files found: {len(files)}")

results = []
for full_path, folder, filename in files:
    ext = os.path.splitext(filename)[1].lower()
    clean_folder = folder.replace("\\", "_").replace(" ", "_").replace(".", "root")
    thumb_name = f"{clean_folder}_{filename}.jpg"
    thumb_path = os.path.join(preview_dir, thumb_name)
    
    if ext in [".heic", ".png", ".jpg"]:
        cmd = f'ffmpeg -y -i "{full_path}" -vf "scale=480:-1" -vframes 1 "{thumb_path}"'
    elif ext in [".mov", ".mp4"]:
        cmd = f'ffmpeg -y -ss 00:00:01 -i "{full_path}" -vf "scale=480:-1" -vframes 1 "{thumb_path}"'
    else:
        cmd = None
        
    if cmd:
        subprocess.run(cmd, shell=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
    
    probe_cmd = f'ffprobe -v error -select_streams v:0 -show_entries stream=width,height,duration,nb_frames,display_aspect_ratio -show_entries format=duration,size -of json "{full_path}"'
    p = subprocess.run(probe_cmd, shell=True, capture_output=True, text=True)
    meta = {}
    try:
        data = json.loads(p.stdout)
        s = data.get("streams", [{}])[0]
        fmt = data.get("format", {})
        meta["width"] = s.get("width")
        meta["height"] = s.get("height")
        dur = s.get("duration") or fmt.get("duration")
        meta["duration"] = round(float(dur), 2) if dur else 0
        meta["size_mb"] = round(int(fmt.get("size", 0)) / (1024 * 1024), 2)
    except Exception as e:
        meta["error"] = str(e)
        
    results.append({
        "file": filename,
        "folder": folder,
        "path": full_path,
        "meta": meta,
        "thumb": thumb_name
    })

with open("media_inventory.json", "w", encoding="utf-8") as f:
    json.dump(results, f, indent=2)

print("Analysis & thumbnails generation complete!")

File: scripts/analyze_media.js
const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

const baseDir = 'C:\\Users\\javi_\\Desktop\\Proyectos\\UtopiaVanLife\\Landing Videos-images\\Landing';
const previewDir = 'C:\\Users\\javi_\\Desktop\\Proyectos\\UtopiaVanLife\\UtopiaVanLife_appWb\\temp_previews';

if (!fs.existsSync(previewDir)) {
  fs.mkdirSync(previewDir, { recursive: true });
}

function getFiles(dir, rel = '') {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat && stat.isDirectory()) {
      results = results.concat(getFiles(fullPath, path.join(rel, file)));
    } else {
      results.push({ fullPath, relFolder: rel, file });
    }
  });
  return results;
}

const files = getFiles(baseDir);
console.log(`Found ${files.length} files.`);

const inventory = [];

for (const item of files) {
  const ext = path.extname(item.file).toLowerCase();
  const cleanFolder = item.relFolder.replace(/\\/g, '_').replace(/ /g, '_') || 'root';
  const thumbName = `${cleanFolder}_${item.file}.jpg`;
  const thumbPath = path.join(previewDir, thumbName);

  let thumbGenerated = false;
  try {
    if (['.heic', '.png', '.jpg', '.jpeg'].includes(ext)) {
      execSync(`ffmpeg -y -i "${item.fullPath}" -vf "scale=480:-1" -vframes 1 "${thumbPath}"`, { stdio: 'ignore' });
      thumbGenerated = true;
    } else if (['.mov', '.mp4'].includes(ext)) {
      execSync(`ffmpeg -y -ss 00:00:01 -i "${item.fullPath}" -vf "scale=480:-1" -vframes 1 "${thumbPath}"`, { stdio: 'ignore' });
      thumbGenerated = true;
    }
  } catch (err) {
    console.error(`Error generating thumb for ${item.file}:`, err.message);
  }

  let meta = {};
  try {
    const probeOut = execSync(`ffprobe -v error -select_streams v:0 -show_entries stream=width,height,duration -show_entries format=duration,size -of json "${item.fullPath}"`, { encoding: 'utf-8' });
    const probeData = JSON.parse(probeOut);
    const stream = (probeData.streams && probeData.streams[0]) || {};
    const format = probeData.format || {};
    meta.width = stream.width;
    meta.height = stream.height;
    const dur = stream.duration || format.duration;
    meta.duration = dur ? parseFloat(dur).toFixed(2) : '0';
    meta.sizeMb = (parseInt(format.size || 0, 10) / (1024 * 1024)).toFixed(2);
  } catch (err) {
    meta.error = err.message;
  }

  inventory.push({
    file: item.file,
    relFolder: item.relFolder,
    fullPath: item.fullPath,
    ext,
    meta,
    thumbName,
    thumbGenerated
  });
}

fs.writeFileSync('media_inventory.json', JSON.stringify(inventory, null, 2));
console.log('Finished analyzing media!');
